tests/test_rio_blocks.py

```diff
@@ -0,0 +1,132 @@
+"""Unittests for $ rio blocks"""
+
+
+import itertools as it
+import json
+import os
+import re
+
+import numpy as np
+import six
+
+import rasterio
+from rasterio.warp import transform_bounds
+from rasterio.rio.main import main_group
+
+
+def check_features_block_windows(features, src, bidx):
+
+    """Compare GeoJSON features to a datasource's blocks + windows.
+
+    Parameters
+    ----------
+    features : iter
+        GeoJSON features.
+    src : RasterReader
+        Open input datasource.
+    """
```

Comment:
@geowurster if you are going all-in with docstrings, please add the `Returns` part too.